### ISSUE TITLE
Extremely high GPU temperature reported by dcgm-exporter

### ISSUE THREAD

**Issue by age9990** [NVIDIA/dcgm-exporter#312]
### What is the version?

3.3.5-3.4.0

### What happened?

After upgrading from 3.1.3-3.1.2 to 3.3.5-3.4.0, GPU temperature metric DCGM_FI_DEV_GPU_TEMP occationally reports extremely large number, ex: 345, 82505, 200000, 644245923. It never happened when using 3.1.3-3.1.2 version.

### What did you expect to happen?

Correct GPU temperature is returned.

### What is the GPU model?

A100, H100

### What is the environment?

K8S v1.23, v1.24

### How did you deploy the dcgm-exporter and what is the configuration?

Default setting come from gpu-operator v23.9.2

**Comment by nvvfedorov** [comment]
@age9990. Thank you for reporting the bug.  Can you run the dcgm-exporter in debug mode and share logs? 
Here is how you can do this:
```
dcgm-exporter --debug true --enable-dcgm-log true --dcgm-log-level DEBUG
```

**Comment by age9990** [comment]
Updated to 3.3.5-3.4.1, and enabled debug mode and log, the issue still happened.

time="2024-04-14T20:33:16Z" level=info msg="Appended entity i64 eg 1, eid 1, fieldId 150, ts 1713126796036793 , value1 4278190080, value2 0, cached 1, buffered 0 [/workspaces/dcgm-rel_dcgm_3_3-postmerge/dcgmlib/src/DcgmCacheManager.cpp:6262] [DcgmCacheManager::AppendEntityInt64]" dcgm_level=DEBUG
time="2024-04-14T20:33:16Z" level=info msg="Preparing to update watchInfo 0x7f9c34005fe0, eg 1, eid 1, fieldId 150 [/workspaces/dcgm-rel_dcgm_3_3-postmerge/dcgmlib/src/DcgmCacheManager.cpp:5277] [DcgmCacheManager::ActuallyUpdateAllFields]" dcgm_level=DEBUG

**Comment by jz543fm** [comment]
I've encountered the same problem with the dcgm-exporter version: `3.3.3-3.3.0`, so I upgraded version to `3.3.5-3.4.1` and enabled debug options as you mentioned above...., I can not see the `EnableDCGMLog: true` because it is on false, I used args you mentioned above: `dcgm-exporter --debug true --enable-dcgm-log true --dcgm-log-level DEBUG` 

Before this I've created my own system logger (systemd service) for temperature with the usage of `nvidia-smi`, `while true; do nvidia-smi -q -d temperature; done > /var/log/templog.log` and I found out that temperature was normal from `nvidia-smi` and `dcgm-exporter` reported different temperature values, when I will have log from dcgm-exporter I will paste it here later

```bash
2024/04/23 09:21:28 maxprocs: Leaving GOMAXPROCS=64: CPU quota undefined                                                                                                                                         │
│ time="2024-04-23T09:21:28Z" level=info msg="Starting dcgm-exporter"                                                                                                                                              │
│ time="2024-04-23T09:21:28Z" level=debug msg="Debug output is enabled"                                                                                                                                            │
│ time="2024-04-23T09:21:28Z" level=debug msg="Command line: dcgm-exporter --debug true --enable-dcgm-log true --dcgm-log-level DEBUG"                                                                             │
│ time="2024-04-23T09:21:28Z" level=debug msg="Loaded configuration" dump="&{CollectorsFile:/etc/dcgm-exporter/default-counters.csv Address::9400 CollectInterval:30000 Kubernetes:true KubernetesGPUIdType:uid Co │
│ llectDCP:true UseOldNamespace:false UseRemoteHE:false RemoteHEInfo:localhost:5555 GPUDevices:{Flex:true MajorRange:[] MinorRange:[]} SwitchDevices:{Flex:true MajorRange:[] MinorRange:[]} CPUDevices:{Flex:true │
│  MajorRange:[] MinorRange:[]} NoHostname:false UseFakeGPUs:false ConfigMapData:none MetricGroups:[] WebSystemdSocket:false WebConfigFile: XIDCountWindowSize:300000 ReplaceBlanksInModelName:false Debug:true Cl │
│ ockEventsCountWindowSize:300000 EnableDCGMLog:false DCGMLogLevel:NONE PodResourcesKubeletSocket:/var/lib/kubelet/pod-resources/kubelet.sock}"    
```

Tested on K8s: `v1.29.0`

**Comment by age9990** [comment]
Tested 3.3.0-3.2.0 & 3.1.6-3.1.3, same problem. Rollback to use 3.1.3-3.1.2, hope this issue can be fixed soon.

**Comment by age9990** [comment]
@nvvfedorov new DCGM 3.3.6 release is out and fixes this issue. Hope new version of DCGM Exporter will be released ASAP, much appreciated.

**Comment by nvvfedorov** [comment]
@age9990 , The new version of the DCGM-exporter has been released: 3.3.6-3.4.2
